### DISCUSSION TITLE
How to have snippets inheritance when using `from_lua` loader?

### DISCUSSION THREAD

**Discussion by bew** [L3MON4D3/LuaSnip#1003] [Q&A] [answered]
Hello, I'm starting to use luasnip (:sparkles: _it's gonna be amaaazing!_)
(rewriting all of my Ultisnips snippets incrementally)

I'm using the `from_lua` loader to load all my snippets.

In ultisnips I had both `sh.snippets` & `zsh.snippets` where the latter would extend from the former (with a lower priority).

Is there a similar way to do this with Luasnips while using `from_lua` loader?
I didn't find anything like this mentioned [in the docs](https://github.com/L3MON4D3/LuaSnip/blob/master/DOC.md#lua)

I was thinking I could get all sh snippets and append them to the zsh snippets table, but that would make the snippets loading order-dependant, if it worked, it wouldn't properly live reload in zsh files if I change sh snippets.

**Comment by L3MON4D3** [comment]
Oh damn, I've seen you in so many discussion and had just assumed you're using luasnip since forever..
Hope you'll enjoy it! :D

I think one good way of extending specific files (as opposed to filetypes) is giving them uncommon names (maybe `my_sh.lua` and `my_zsh.lua`), and use `ls.filetype_extend("zsh", {"my_zsh", "my_sh"})` and `ls.filetype_extend("sh", {"my_sh"})`.
You can influence the relative priority (look into `my_zsh` or `my_sh` first) that way as well, I believe the one that is specified first will also have a higher priority (so with the above call, a snippet in `my_zsh` will be expanded before one in `my_sh`) BUT!! this is an implementation detail, although pretty unlikely to change.

**Reply by bew** [reply, marked as answer]
Thanks that sent me in the right direction, I ended up with this in my `zsh.lua`:
```lua
ls.filetype_set("zsh", {"zsh", "sh"})
```

---
> Oh damn, I've seen you in so many discussion and had just assumed you're using luasnip since forever..
> Hope you'll enjoy it! :D

Ahah yeah I've been watching luasnip for a good while, but I had other parts of my config to take care of properly before diving in!
(I still haven't enable treesitter & lsp stuff 👀 They are my next things once I'm more confident with luasnip!)

I noticed a few bugs with extmarks while navigating placeholders it seems, expect a few issues in the future (or request for better understanding)

**Reply by L3MON4D3** [reply]
Ah, yeah of course, that is even more straightforward :D

> Ahah yeah I've been watching luasnip for a good while, but I had other parts of my config to take care of properly before diving in!
(I still haven't enable treesitter & lsp stuff 👀 They are my next things once I'm more confident with luasnip!)

Daang, you got some good things coming, they are both a lot of fun to play with :)
I'd hoped we ironed out all those extmark-issues, definitely let me know about those :+1:
